File: src/plugin/inter/JewellerySmithingInterfacePlugin.java
package plugin.inter;

import com.rs.game.content.skills.crafting.JewelrySmithing;
import com.rs.game.entity.actor.player.Player;
import com.rs.game.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class JewellerySmithingInterfacePlugin extends InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		JewelrySmithing.handleButtonClick(player, componentId, packetId == 14 ? 1 : packetId == 67 ? 5 : 10);
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(675);
	}
}

File: src/main/kotlin/plugin/rsinterface/JewellerySmithingInterfacePlugin.kt
package plugin.rsinterface;

import org.redrune.game.content.entity.actor.player.skills.crafting.JewelrySmithing;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/31/2017
 */
public class JewellerySmithingInterfacePlugin implements InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		JewelrySmithing.handleButtonClick(player, componentId, packetId == 14 ? 1 : packetId == 67 ? 5 : 10);
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(675);
	}
}
